### ISSUE TITLE
🐛[BUG]: Healpix dataloaders use val_dataset for test

### ISSUE THREAD

**Issue by anna-dodson** [NVIDIA/physicsnemo#988]
### Version

1.1.0

### On which installation method(s) does this occur?

Pip, Docker

### Describe the issue

In datapipes.healpix.data_modules.py:

    def test_dataloader(self, num_shards=1, shard_id=0) -> DataLoader:
        """Setup the test dataloader

        Parameters
        ----------
        num_shards: int, optional
            The total total number of distributed shards
            default is 1 meaning distributed test is not being used
        shard_id: int, optional
            The shard number of this instance of the dataloader, default 0

        Returns
        -------
        DataLoader: The test dataloader
        """
        sampler = None
        if num_shards > 1:
            sampler = DistributedSampler(
                **self.val_dataset,**
                num_replicas=num_shards,
                rank=shard_id,
                shuffle=False,
                drop_last=False,
            )

        loader = DataLoader(
            dataset=self.test_dataset,
            pin_memory=self.pin_memory,
            num_workers=self.num_workers,
            shuffle=False,
            drop_last=False,
            sampler=sampler,
            collate_fn=self.collate_fn,
            batch_size=self.dataloader_batch_size,
        )

DistributedSampler uses **self.val_dataset** instead of **self.test_dataset** for distributed loading, this is undesired behavior I imagine.

### Minimum reproducible example

```shell

```

### Relevant log output

```shell

```

### Environment details

```shell

```
